### PR TITLE
Mitra/Refactor `queryParamData` into `usePlatformQueryParams` custom hook

### PR DESCRIPTION
**Pull request by mitra-deriv**
Changes:

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/common/utility.js [hunk 1]
```diff
@@ -472,15 +473,15 @@ export const handleRedirect = (residence, current_client_country) => {
...
-        const platform_list = ['derivgo', 'p2p']
-        return platform_list.includes(platform_name) ? platform_name : ''
-    } else return ''
-}
-
 export const isLocalhost = () => !!(isBrowser() && process.env.NODE_ENV === 'development')
 
 export const isTestlink = () => !!(isBrowser() && window.location.hostname.includes('binary.sx'))
+
+export const usePlatformQueryParam = () => {
```
Comment (farzin-fs) [inline src/common/utility.js:480]: @mitra-deriv Would you please extract this hook into a separate file under `src/components/hooks` 🙏🏻

File: src/components/hooks/use-platform-query-param.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { useQueryParam, StringParam } from 'use-query-params'
+
+export const usePlatformQueryParam = () => {
+    const [platform] = useQueryParam('platform', StringParam)
+    const isDerivGo = platform === 'derivgo'
+    const isDerivP2P = platform === 'p2p'
+    const hasPlatform = isDerivGo || isDerivP2P
```
Comment (yashim-deriv) [inline src/components/hooks/use-platform-query-param.tsx:7]:
```suggestion
    const is_deriv_go = platform === 'derivgo'
    const is_derivp2p = platform === 'p2p'
    const has_platform = is_deriv_go || is_derivp2p
```

File: src/pages/help-centre/index.tsx [hunk 3]
```diff
@@ -260,7 +262,7 @@ const HelpCentre = () => {
             <Desktop breakpoint={'tabletS'}>
                 <Community />
             </Desktop>
-            <DidntFindYourAnswerBanner />
+            {!isDerivGo && <DidntFindYourAnswerBanner />}
```
Comment (yashim-deriv) [inline src/pages/help-centre/index.tsx:265]:
```suggestion
            {isDerivGo && <DidntFindYourAnswerBanner />}
```
No need to coerce if you can guarantee that it is a boolean

**Review by yashim-deriv — COMMENTED** [review]
Some naming convention comments

**Review by ashraf-deriv — DISMISSED** [review]

File: src/pages/help-centre/index.tsx (hunk 3 above)
Comment (mitra-deriv) [reply src/pages/help-centre/index.tsx]: As discussed in person, We need to hide this section only for DerivGO users so we need to reverse the condition here🙏

**Review by farzin-fs — DISMISSED** [review]

**Review by yashim-deriv — APPROVED** [review]

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**